lib/backgroundscanner.php

```diff
@@ -42,89 +49,50 @@ public static function check() {
+	 */
+	protected function initFS(){
+		//get a random user (needed to mount FS)
+		$userManager = \OC_User::getManager();
+		$results = $userManager->search('', 2, 0);
+		$anyUser = array_pop($results);
 
-	/*
-	* This function is a hack, it doesn't work if the $storage_id is a hash.
-	*/
-	protected static function getStorage($serverContainer, $storage_id) {
-		if (strpos($storage_id, 'local::') === 0) {
-			$arguments = array(
-				'datadir' => substr($storage_id, 7),
-			);
-			return new \OC\Files\Storage\Local($arguments);
-		}
-		if (strpos($storage_id, 'home::') === 0) {
-			$userid = substr($storage_id, 6);
-			$user = $serverContainer->getUserManager()->get($userid);
-			$arguments = array(
-				'user' => $user,
-			);
-			return new \OC\Files\Storage\Home($arguments);
-		}
-		return null;
+		\OC_Util::tearDownFS();
+		\OC_Util::setupFS($anyUser->getUID());
 	}
 
-	public static function scan($id, $path, $storage) {
-		$fileStatus = \OCA\Files_Antivirus\Scanner::scanFile($storage, $path);
-		$result = $fileStatus->getNumericStatus();
-		
-		//TODO: Fix undefined $user here
-		switch($result) {
-			case \OCA\Files_Antivirus\Status::SCANRESULT_UNCHECKED:
-				\OCP\Util::writeLog('files_antivirus', 'File "'.$path.'" with id "'.$id.'": is not checked', \OCP\Util::ERROR);
-				break;
-			case \OCA\Files_Antivirus\Status::SCANRESULT_INFECTED:
-				$infected_action = \OCP\Config::getAppValue('files_antivirus', 'infected_action', 'only_log');
-				if ($infected_action == 'delete') {
-					\OCP\Util::writeLog('files_antivirus', 'File "'.$path.'" with id "'.$id.'": is infected, file deleted', \OCP\Util::ERROR);
-					$storage->unlink($path);
-				}
-				else {
-					\OCP\Util::writeLog('files_antivirus', 'File "'.$path.'" with id "'.$id.'": is infected', \OCP\Util::ERROR);
-				}
-				break;
-			case \OCA\Files_Antivirus\Status::SCANRESULT_CLEAN:
-				try {
-					$stmt = \OCP\DB::prepare('DELETE FROM `*PREFIX*files_antivirus` WHERE `fileid` = ?');
-					$result = $stmt->execute(array($id));
-					if (\OCP\DB::isError($result)) {
-						\OCP\Util::writeLog('files_antivirus', __METHOD__. ', DB error: ' . \OC_DB::getErrorMessage($result), \OCP\Util::ERROR);
-						return;
-					}
-					$stmt = \OCP\DB::prepare('INSERT INTO `*PREFIX*files_antivirus` (`fileid`, `check_time`) VALUES (?, ?)');
-					$result = $stmt->execute(array($id, time()));
-					if (\OCP\DB::isError($result)) {
-						\OCP\Util::writeLog('files_antivirus', __METHOD__. ', DB error: ' . \OC_DB::getErrorMessage($result), \OCP\Util::ERROR);
-						return;
-					}
-				} catch(\Exception $e) {
-					\OCP\Util::writeLog('files_antivirus', __METHOD__.', exception: '.$e->getMessage(), \OCP\Util::ERROR);
-				}
-				break;
-		}
+
+	/**
+	 * Get a mimetypeId for httpd/unix-directory
+	 * @return int
+	 */
+	protected function getDirectoryMimetype(){
+		$query = \OCP\DB::prepare('SELECT `id` FROM `*PREFIX*mimetypes` WHERE `mimetype` = ?');
```

Comment:
hmm ... we should have a core function to get such information @icewind1991 any idea? THX